Core/Executor/ContentTypeManager.php

```diff
@@ -284,6 +285,25 @@ protected function delete($step)
     }
 
     /**
+     * Method that create a content type if it doesn't already exist, or update it if it do.
+     */
+    protected function upsert($step)
+    {
+        if (!isset($step->dsl['identifier'])) {
```

Comment:
Interesting point here... do we want to only allow matching by identifier for upserts ? what if the user also has put in a 'match' clause ?

Reply:
The create mode needs the identifier and looks not after match (https://github.com/datafactory/ezmigrationbundle/blob/99f339714dfef17b27a6e6d6ad9c5695e115e712/Core/Executor/ContentTypeManager.php#L49)
The update mode looks after identifier and match.

So I think we should only look after identifier..

Reply:
But Yes: If match is set, a exception will be helpful